src/shared/bg_pmove.cpp

```diff
@@ -3603,9 +3598,65 @@ PM_Weapon
 Generates weapon events and modifies the weapon counter
 ==============
 */
+
+// for weapons with long, noticeable cooldown (mantis pounce)
+static void cooldown( pml_t & pml )
+{
+	pm->ps->weaponCharge -= pml.msec;
+	if ( pm->ps->weaponCharge < 0 )
+	{
+		pm->ps->weaponCharge = 0;
+	}
+}
+
+// charge the skill up to max by amount.
+// If the skill was at maximum and trigger is true, multiply weaponCharge
+// but mult.
+// Triggers a warning if warning is reached
+// Return true if skill was already at maximum value
+static bool chargeSkill( int max, int amount, int warning, bool trigger, float mult )
```

Comment:
The idea of this function is hard to understand - a real head-scratcher. I think it would be better to just apply min/max/clamp more in the original code. Same goes for dischargeSkill

Reply:
To be honest, the way the charging of weapons is done in code is not intuitive neither, and I refrained myself to make it more generic, since it would be out of the scope of this PR (it would explicitly affect gameplay).

I agree that this is suboptimal, though. Maybe it would be better to just drop this commit for now, until I (or someone else) can come with a better solution?

Reply:
Dropping the commit works for me

Reply:
Done.